src/Core/Content/Property/PropertyGroupCollection.php

```diff
@@ -47,20 +47,30 @@ public function sortByPositions(): void
 
     public function sortByConfig(): void
     {
-        /** @var Entity $group */
         foreach ($this->elements as $group) {
-            $options = $group->get('options');
-            if (!$options instanceof EntityCollection) {
+            $options = $group->getOptions();
+            if (!$options instanceof PropertyGroupOptionCollection) {
                 continue;
             }
 
-            $options->sort(static function (Entity $a, Entity $b) use ($group) {
-                if ($group->get('sortingType') === PropertyGroupDefinition::SORTING_TYPE_ALPHANUMERIC) {
-                    return strnatcmp((string) $a->getTranslation('name'), (string) $b->getTranslation('name'));
+            $columns = [];
+            $entities = [];
+
+            $sortingType = PropertyGroupDefinition::SORTING_TYPE_ALPHANUMERIC;
+
+            foreach ($options->getIterator() as $option) {
+                if ($sortingType) {
+                    $columns[] = (string) ($option->getTranslation('name') ?? '');
+                } else {
+                    $columns[] = (int) ($option->getTranslation('position') ?? $option->getPosition() ?? 0);
                 }
 
-                return ($a->getTranslation('position') ?? $a->get('position') ?? 0) <=> ($b->getTranslation('position') ?? $b->get('position') ?? 0);
-            });
+                $entities[] = $option;
+            }
+
+            array_multisort($columns, \SORT_ASC, $entities);
```

Comment:
with this, you do not have `strnatcmp` behaviour anymore, right?

Reply:
fixed it with `\SORT_NATURAL` flag for multisort, we had tests that covered this ;)